version: '3.9'
services:
  service-mysql:
    image: mysql:8.0.34-debian
    restart: unless-stopped
    environment:
      - MYSQL_DATABASE=cards
      - MYSQL_ROOT_PASSWORD=mauFJcuf5dhRMQrjj
      - MYSQL_USER=user
      - MYSQL_PASSWORD=mauFJcuf5dhRMQrjj
    ports:
      - '3308:3306'
    volumes:
      - cards:/var/lib/mysql
  cards-service:
    image: cards-service:latest
    platform: linux/amd64
    build:
      context: .
      dockerfile: Dockerfile
      cache_from:
        - openjdk:20-slim
        - maven:3.9.3-amazoncorretto-20
    ports:
      - target: 8080
        published: 8080
    volumes:
      - "./src/main/resources/:/datadir/"
      - "${HOME}/.m2:/root/.m2"
    depends_on:
      - service-mysql
    environment:
      - DATABASE_URL=jdbc:mysql://service-mysql:3306/cards?allowPublicKeyRetrieval=true&useSSL=false&useLegacyDatetimeCode=false
      - LIQUIBASE_DATABASE_URL=jdbc:mysql://service-mysql:3306/cards?&createDatabaseIfNotExist=true&allowPublicKeyRetrieval=true&useSSL=false&useLegacyDatetimeCode=false
      - LIQUIBASE_CHANGELOG=db/changelog/db-changelog.yaml
      - DATABASE_NAME=cards
      - DATABASE_USER=user
      - DATABASE_PASSWORD=mauFJcuf5dhRMQrjj
      - CONNECTION_POOL_ENABLED=true
      - CONNECTION_POOL_INITIAL_SIZE=10
      - MIN_POOL_SIZE=3
      - MAX_POOL_SIZE=20
      - VALIDATION_DEPTH=local
      - VALIDATION_QUERY=SELECT 1
      - MAX_IDLE_TIME=3m
      - HASH_FUNCTION=SHA3-256
      - LOG_PATH=/var/log/
      - LOG_LEVEL=DEBUG
      - PORT=8080
      - JWT_TOKEN=8705E66163F4715FA42586D510D859BEFDBBBFBE52A191A12E3B7D2EDD230854
    external_links:
      - service-mysql
    restart: on-failure
  # Names our volume
volumes:
  cache:
    driver: local
  cards-service:
  cards: